management-cluster-access/README.md

```diff
@@ -0,0 +1,119 @@
+- A long living client-certificate granting admin access is a security risk since permissions can not be revoked.
+- We do not want this to be the primary authentication method for human users. (That should remain SSO)
+- If we want to support this, we need to decide whether we want to use vault or something else.
+- Whichever method we choose, we need to ensure that TTL duration is limited to minimize the security risk. 
+
+### Storing kubeconfig in LastPass
+
+In the CAPI product, a kubeconfig is stored in LastPass as an emergency fallback, during the bootstrap process.
+
+- This method of authentication is completely separate from SSO and does not have shared dependencies.
+- It is not integrated in our tooling and we need to pull the kubeconfig using lastpass cli.
+- As it is right now, access is neither easily revokable nor is there a rotation. 
+- In case of a security threat this implies rotating the api server CA and all certificates.
+- Since it is already integrated, it makes sense to keep it for emergencies while there is no alternative.
+- If we want to support this for a longer time we need to improve security.
+- We could support this as fallback method in `opsctl login` by managing access to lastpass.com or calling the `lastpass` CLI to ease operations.
+
+### Introduce a second identity provider for SSO
+
+Dex supports a wide array of identity providers. Customers are already using many of them. At the present moment, giant swarm staff can only login using github.
+However, we could add more providers.
+
+- We favor SSO when it comes to identity management for human engineers.
+- Could be introduced for workload cluster access as well.
+- Using e.g. azure AD would make sense since customers are using it.
+- We might be able to simplify account automation when using another identity provider than github.
+- We would still have a dependency on dex and this method will not completely remove the single point of failure.
+- Adding another provider might mean individual setup for each existing configuration. (e.g. in github individual apps are needed) We should avoid this if possible since it already does not scale.
+- While dex allows an array of connectors, we currently use a fixed naming for our single `giantswarm` connector. We need to revisit what relies on this convention to ensure the second connector is equal.
+- Regardless of other means auf authentication being implemented, adding another dex idp would be beneficial.
+
+### Service account tokens
+
+Other than client certificates, we could also look at service accounts as a means of managing access.
+
+- Service accounts can be rotated so they carry less risk in that regard than client certificates.
+- This should not be a standard access method for human engineers.
+- We would need to discover how to implement this. A dedicated controller for managing service account access is thinkable.
+
+### Other access methods
+
+- To be discussed
```

Comment:
Creating a controller (running on each MC) which creates a client-certificate per existing cluster (from the in-management cluster stored CA) and write this secret back to `lastpass` (e.g. via [`lastpass-go`(https://github.com/ansd/lastpass-go)]

Pros:
* TTL of the client-certificate could be configurable
* no dependency to a running management-cluster for accessing a workload cluster
* Accessing the management-cluster via that client-certificate is also possible

> **Note**
> * does lastpass has support for "technical users" (via tokens)
> * we have to check of we are able to create some kind of folder per installation

Reply:
So basically, the controller would ensure that this certificate is rotated often, right?
It would be a big improvement in comparison to just creating it on bootstrap.

Reply:
> So basically, the controller would ensure that this certificate is rotated often, right?

I want change anything for the cluster bootstrap/lifecycle. I just thought about creating a "short-lived" client-certificate which could used to authenticate against a kubernetes API. And to remove the runtime-dependency to a management-cluster, we could sync (periodically) this client-certificate back to `keepass`. (And as a management-cluster is from a CAPI point of view similar to a workload cluster, this will also give us "emergency" access to the management cluster)